complement/pipeline.yml

```diff
@@ -37,3 +40,57 @@ steps:
       automatic:
         - exit_status: 128
           limit: 3
+
+  - command:
+      # Build the Synapse for Complement docker image, which is located at:
+      # https://github.com/matrix-org/complement/blob/master/dockerfiles/Synapse.Dockerfile.
+      - docker build -t complement-synapse -f dockerfiles/Synapse.Dockerfile dockerfiles/
+      # Run the tests!
+      - COMPLEMENT_BASE_IMAGE=complement-synapse go test -v -tags "synapse_blacklist" ./tests
+    label: "\U0001F9EA Complement / Synapse Monolith / :go: 1.15"
+    agents:
+      queue: "medium"
+    plugins:
+      - docker#v3.7.0:
+          # The dockerfile for this image is at https://github.com/matrix-org/complement/blob/master/dockerfiles/ComplementCIBuildkite.Dockerfile.
+          image: "matrixdotorg/complement:latest"
+          mount-buildkite-agent: false
+          # Complement needs to know if it is running under CI
+          environment:
+           - "CI=true"
+          publish: [ "8448:8448" ]
+          # Complement uses Docker so pass through the docker socket. This means Complement shares
+          # the hosts Docker.
+          volumes:
+            - "/var/run/docker.sock:/var/run/docker.sock"
```

Comment:
this will give access to run arbitrary executables on the build server, so will mean that we can't enable CI for third-party PRs. possibly we could allow it if we ran the build on one of the burn-after-build agents, but we should probably discuss that in CI internal first.

Reply:
We had a quick chat internally and have opted for using burn-after-build agents, which needed only a quick change made in cf896c8fd797130d60733754821790b90d0ab585.